# Pré processamento dos dados


In [1]:
import numpy as np
import pandas as pd

dados = pd.read_csv('../tratado.csv') 

dados.head()
dados.shape

(746, 12)

## Tranformação das variaveis categoricas nominais

In [2]:
# fazer uma copia do dataset, para trabalhar sem riscos
dados2 = pd.DataFrame.copy(dados)

In [3]:
dados2.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [4]:
# fazer alterações nas colunas de nominais para ordinais
dados2['Sex'].replace({'M': 0, 'F': 1}, inplace=True)
dados2['ChestPainType'].replace({'TA': 0, 'ATA': 1, 'NAP': 2, 'ASY': 3,}, inplace=True)
dados2['RestingECG'].replace({'Normal': 0, 'ST': 1, 'LVH': 2}, inplace=True)
dados2['ExerciseAngina'].replace({'N': 0, 'Y': 1}, inplace=True)
dados2['ST_Slope'].replace({'Up': 0, 'Flat': 1, 'Down': 2}, inplace=True)

C:\Windows\Temp\ipykernel_17240\797369926.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dados2['Sex'].replace({'M': 0, 'F': 1}, inplace=True)
C:\Windows\Temp\ipykernel_17240\797369926.py:2: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  dados2['Sex'].replace({'M': 0, 'F': 1}, inplace=True)
C:\W

In [5]:
dados2.dtypes

Age                 int64
Sex                 int64
ChestPainType       int64
RestingBP           int64
Cholesterol         int64
FastingBS           int64
RestingECG          int64
MaxHR               int64
ExerciseAngina      int64
Oldpeak           float64
ST_Slope            int64
HeartDisease        int64
dtype: object

In [6]:
# Selecioanr colunas relevantes para classificação
previsores = dados2.iloc[:, 0:11].values

In [7]:
previsores

array([[40. ,  0. ,  1. , ...,  0. ,  0. ,  0. ],
       [49. ,  1. ,  2. , ...,  0. ,  1. ,  1. ],
       [37. ,  0. ,  1. , ...,  0. ,  0. ,  0. ],
       ...,
       [57. ,  0. ,  3. , ...,  1. ,  1.2,  1. ],
       [57. ,  1. ,  1. , ...,  0. ,  0. ,  1. ],
       [38. ,  0. ,  2. , ...,  0. ,  0. ,  0. ]])

In [8]:
previsores.shape

(746, 11)

In [9]:
#meu alvo == target
alvo = dados.iloc[:, 11].values

In [10]:
alvo

array([0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0,
       0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1,
       1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0,
       1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1,
       1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0,

## Escalocamento

In [11]:
# estatisticas
dados2.describe()

Age         Sex  ChestPainType   RestingBP  Cholesterol  \
count  746.000000  746.000000     746.000000  746.000000   746.000000   
mean    52.882038    0.243968       2.163539  133.022788   244.635389   
std      9.505888    0.429762       0.955414   17.282750    59.153524   
min     28.000000    0.000000       0.000000   92.000000    85.000000   
25%     46.000000    0.000000       1.000000  120.000000   207.250000   
50%     54.000000    0.000000       2.000000  130.000000   237.000000   
75%     59.000000    0.000000       3.000000  140.000000   275.000000   
max     77.000000    1.000000       3.000000  200.000000   603.000000   

        FastingBS  RestingECG       MaxHR  ExerciseAngina     Oldpeak  \
count  746.000000  746.000000  746.000000      746.000000  746.000000   
mean     0.167560    0.639410  140.226542        0.384718    0.901609   
std      0.373726    0.838664   24.524107        0.486855    1.072861   
min      0.000000    0.000000   69.000000        0.000000   -0.100000   
25%      0.000000    0.000000  122.000000        0.000000    0.000000   
50%      0.000000    0.000000  140.000000        0.000000    0.500000   
75%      0.000000    1.000000  160.000000        1.000000    1.500000   
max      1.000000    2.000000  202.000000        1.000000    6.200000   

         ST_Slope  HeartDisease  
count  746.000000    746.000000  
mean     0.589812      0.477212  
std      0.598076      0.499816  
min      0.000000      0.000000  
25%      0.000000      0.000000  
50%      1.000000      0.000000  
75%      1.000000      1.000000  
max      2.000000      1.000000

In [12]:
# usar padronização ou normalização para ajustar desvios absurdos
# Padronização (utilizar média e descio como ref.)
#  Normalização (utilizar os valores máximo e mínimo como ref.)

from sklearn.preprocessing import StandardScaler # padronização

In [13]:
# vai transformar previsores com relação a sua scscala
previsores_esc = StandardScaler().fit_transform(previsores)

In [14]:
previsores_esc

array([[-1.35607325, -0.56806253, -1.21865408, ..., -0.79074163,
        -0.84094155, -0.9868448 ],
       [-0.40865641,  1.76036959, -0.17128548, ..., -0.79074163,
         0.0917709 ,  0.6863057 ],
       [-1.67187886, -0.56806253, -1.21865408, ..., -0.79074163,
        -0.84094155, -0.9868448 ],
       ...,
       [ 0.43349189, -0.56806253,  0.87608311, ...,  1.26463557,
         0.27831339,  0.6863057 ],
       [ 0.43349189,  1.76036959, -1.21865408, ..., -0.79074163,
        -0.84094155,  0.6863057 ],
       [-1.56661032, -0.56806253, -0.17128548, ..., -0.79074163,
        -0.84094155, -0.9868448 ]])

In [15]:
dados3 = pd.DataFrame(previsores_esc)
dados3

0         1         2         3         4         5         6   \
0   -1.356073 -0.568063 -1.218654  0.403980  0.750494 -0.448652 -0.762927   
1   -0.408656  1.760370 -0.171285  1.561980 -1.093405 -0.448652 -0.762927   
2   -1.671879 -0.568063 -1.218654 -0.175019  0.648995 -0.448652  0.430246   
3   -0.513925  1.760370  0.876083  0.288180 -0.518244 -0.448652 -0.762927   
4    0.117686 -0.568063 -0.171285  0.982980 -0.839657 -0.448652 -0.762927   
..        ...       ...       ...       ...       ...       ...       ...   
741 -0.829731 -0.568063 -2.266023 -1.333019  0.327582 -0.448652 -0.762927   
742  1.591446 -0.568063  0.876083  0.635580 -0.873490  2.228901 -0.762927   
743  0.433492 -0.568063  0.876083 -0.175019 -1.922314 -0.448652 -0.762927   
744  0.433492  1.760370 -1.218654 -0.175019 -0.146081 -0.448652  1.623418   
745 -1.566610 -0.568063 -0.171285  0.288180 -1.177987 -0.448652 -0.762927   

           7         8         9         10  
0    1.296470 -0.790742 -0.840942 -0.986845  
1    0.643613 -0.790742  0.091771  0.686306  
2   -1.722993 -0.790742 -0.840942 -0.986845  
3   -1.314958  1.264636  0.558127  0.686306  
4   -0.743708 -0.790742 -0.840942 -0.986845  
..        ...       ...       ...       ...  
741 -0.335672 -0.790742  0.278313  0.686306  
742  0.031560 -0.790742  2.330281  0.686306  
743 -1.029333  1.264636  0.278313  0.686306  
744  1.378077 -0.790742 -0.840942  0.686306  
745  1.337274 -0.790742 -0.840942 -0.986845  

[746 rows x 11 columns]

In [16]:
dados3.describe()

0             1             2             3             4   \
count  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02   
mean   2.476422e-16  1.904940e-17  1.381082e-16  3.095528e-17 -2.083528e-16   
std    1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00   
min   -2.619296e+00 -5.680625e-01 -2.266023e+00 -2.375219e+00 -2.700473e+00   
25%   -7.244620e-01 -5.680625e-01 -1.218654e+00 -7.540192e-01 -6.324301e-01   
50%    1.176863e-01 -5.680625e-01 -1.712855e-01 -1.750194e-01 -1.291641e-01   
75%    6.440290e-01 -5.680625e-01  8.760831e-01  4.039804e-01  5.136631e-01   
max    2.538863e+00  1.760370e+00  8.760831e-01  3.877979e+00  6.062277e+00   

                 5             6             7             8             9   \
count  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02   
mean   5.714821e-17  3.809881e-17 -3.619387e-16  5.238586e-17  1.714446e-16   
std    1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00   
min   -4.486516e-01 -7.629266e-01 -2.906297e+00 -7.907416e-01 -9.342128e-01   
25%   -4.486516e-01 -7.629266e-01 -7.437078e-01 -7.907416e-01 -8.409415e-01   
50%   -4.486516e-01 -7.629266e-01 -9.243702e-03 -7.907416e-01 -3.745853e-01   
75%   -4.486516e-01  4.302458e-01  8.068275e-01  1.264636e+00  5.581271e-01   
max    2.228901e+00  1.623418e+00  2.520577e+00  1.264636e+00  4.941876e+00   

                 10  
count  7.460000e+02  
mean   9.524702e-17  
std    1.000671e+00  
min   -9.868448e-01  
25%   -9.868448e-01  
50%    6.863057e-01  
75%    6.863057e-01  
max    2.359456e+00

### labelEncoder

In [17]:
# trnasformação de var. categoricas em númericos de forma automatica.
from sklearn.preprocessing import LabelEncoder

In [18]:
dados.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [19]:
previsores2 = dados.iloc[:,0:11].values
previsores2

array([[40, 'M', 'ATA', ..., 'N', 0.0, 'Up'],
       [49, 'F', 'NAP', ..., 'N', 1.0, 'Flat'],
       [37, 'M', 'ATA', ..., 'N', 0.0, 'Up'],
       ...,
       [57, 'M', 'ASY', ..., 'Y', 1.2, 'Flat'],
       [57, 'F', 'ATA', ..., 'N', 0.0, 'Flat'],
       [38, 'M', 'NAP', ..., 'N', 0.0, 'Up']], dtype=object)

In [20]:
previsores2[:,1] = LabelEncoder().fit_transform(previsores2[:,1]) #seleciona o indice da coluna e transforma
previsores2[:,10] = LabelEncoder().fit_transform(previsores2[:,10])
previsores2



array([[40, 1, 'ATA', ..., 'N', 0.0, 2],
       [49, 0, 'NAP', ..., 'N', 1.0, 1],
       [37, 1, 'ATA', ..., 'N', 0.0, 2],
       ...,
       [57, 1, 'ASY', ..., 'Y', 1.2, 1],
       [57, 0, 'ATA', ..., 'N', 0.0, 1],
       [38, 1, 'NAP', ..., 'N', 0.0, 2]], dtype=object)

### OneHotEncoder

In [21]:
# criação de dummys
from sklearn.preprocessing import OneHotEncoder
from sklearn.compose import ColumnTransformer

In [22]:
ColumnTransformer

sklearn.compose._column_transformer.ColumnTransformer

In [23]:
#aplicar transformação
previsores3 = ColumnTransformer(transformers=[('OneHot', OneHotEncoder(), [1,2,6,8,10])], remainder='passthrough').fit_transform(previsores2)

In [24]:
previsores3

array([[0.0, 1.0, 0.0, ..., 0, 172, 0.0],
       [1.0, 0.0, 0.0, ..., 0, 156, 1.0],
       [0.0, 1.0, 0.0, ..., 0, 98, 0.0],
       ...,
       [0.0, 1.0, 1.0, ..., 0, 115, 1.2],
       [1.0, 0.0, 0.0, ..., 0, 174, 0.0],
       [0.0, 1.0, 0.0, ..., 0, 173, 0.0]], dtype=object)

In [25]:
previsores3 = pd.DataFrame(previsores3)
previsores3.head()

0    1    2    3    4    5    6    7    8    9    10   11   12   13  14  \
0  0.0  1.0  0.0  1.0  0.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0  0.0  1.0  40   
1  1.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0  1.0  0.0  49   
2  0.0  1.0  0.0  1.0  0.0  0.0  0.0  0.0  1.0  1.0  0.0  0.0  0.0  1.0  37   
3  1.0  0.0  1.0  0.0  0.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0  1.0  0.0  48   
4  0.0  1.0  0.0  0.0  1.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0  0.0  1.0  54   

    15   16 17   18   19  
0  140  289  0  172  0.0  
1  160  180  0  156  1.0  
2  130  283  0   98  0.0  
3  138  214  0  108  1.5  
4  150  195  0  122  0.0

In [26]:
# vai transformar previsores com relação a sua scscala
previsores3_esc = StandardScaler().fit_transform(previsores3)

In [27]:
previsores3_esc

array([[-0.56806253,  0.56806253, -0.99198919, ..., -0.44865159,
         1.29647027, -0.84094155],
       [ 1.76036959, -1.76036959, -0.99198919, ..., -0.44865159,
         0.64361328,  0.0917709 ],
       [-0.56806253,  0.56806253, -0.99198919, ..., -0.44865159,
        -1.72299329, -0.84094155],
       ...,
       [-0.56806253,  0.56806253,  1.0080755 , ..., -0.44865159,
        -1.02933274,  0.27831339],
       [ 1.76036959, -1.76036959, -0.99198919, ..., -0.44865159,
         1.37807739, -0.84094155],
       [-0.56806253,  0.56806253, -0.99198919, ..., -0.44865159,
         1.33727383, -0.84094155]])

In [28]:
dados4 = pd.DataFrame(previsores3_esc)
dados4

0         1         2         3         4         5         6   \
0   -0.568063  0.568063 -0.991989  1.869218 -0.541197 -0.241156 -0.555673   
1    1.760370 -1.760370 -0.991989 -0.534983  1.847756 -0.241156 -0.555673   
2   -0.568063  0.568063 -0.991989  1.869218 -0.541197 -0.241156 -0.555673   
3    1.760370 -1.760370  1.008076 -0.534983 -0.541197 -0.241156 -0.555673   
4   -0.568063  0.568063 -0.991989 -0.534983  1.847756 -0.241156 -0.555673   
..        ...       ...       ...       ...       ...       ...       ...   
741 -0.568063  0.568063 -0.991989 -0.534983 -0.541197  4.146700 -0.555673   
742 -0.568063  0.568063  1.008076 -0.534983 -0.541197 -0.241156 -0.555673   
743 -0.568063  0.568063  1.008076 -0.534983 -0.541197 -0.241156 -0.555673   
744  1.760370 -1.760370 -0.991989  1.869218 -0.541197 -0.241156  1.799621   
745 -0.568063  0.568063 -0.991989 -0.534983  1.847756 -0.241156 -0.555673   

           7         8         9         10        11        12        13  \
0    0.822438 -0.448652  0.790742 -0.790742 -0.247318 -0.950295  1.066553   
1    0.822438 -0.448652  0.790742 -0.790742 -0.247318  1.052304 -0.937600   
2   -1.215897  2.228901  0.790742 -0.790742 -0.247318 -0.950295  1.066553   
3    0.822438 -0.448652 -1.264636  1.264636 -0.247318  1.052304 -0.937600   
4    0.822438 -0.448652  0.790742 -0.790742 -0.247318 -0.950295  1.066553   
..        ...       ...       ...       ...       ...       ...       ...   
741  0.822438 -0.448652  0.790742 -0.790742 -0.247318  1.052304 -0.937600   
742  0.822438 -0.448652  0.790742 -0.790742 -0.247318  1.052304 -0.937600   
743  0.822438 -0.448652 -1.264636  1.264636 -0.247318  1.052304 -0.937600   
744 -1.215897 -0.448652  0.790742 -0.790742 -0.247318  1.052304 -0.937600   
745  0.822438 -0.448652  0.790742 -0.790742 -0.247318 -0.950295  1.066553   

           14        15        16        17        18        19  
0   -1.356073  0.403980  0.750494 -0.448652  1.296470 -0.840942  
1   -0.408656  1.561980 -1.093405 -0.448652  0.643613  0.091771  
2   -1.671879 -0.175019  0.648995 -0.448652 -1.722993 -0.840942  
3   -0.513925  0.288180 -0.518244 -0.448652 -1.314958  0.558127  
4    0.117686  0.982980 -0.839657 -0.448652 -0.743708 -0.840942  
..        ...       ...       ...       ...       ...       ...  
741 -0.829731 -1.333019  0.327582 -0.448652 -0.335672  0.278313  
742  1.591446  0.635580 -0.873490  2.228901  0.031560  2.330281  
743  0.433492 -0.175019 -1.922314 -0.448652 -1.029333  0.278313  
744  0.433492 -0.175019 -0.146081 -0.448652  1.378077 -0.840942  
745 -1.566610  0.288180 -1.177987 -0.448652  1.337274 -0.840942  

[746 rows x 20 columns]

In [34]:
dados4.describe()

0             1             2           3             4   \
count  7.460000e+02  7.460000e+02  7.460000e+02  746.000000  7.460000e+02   
mean   3.809881e-17  1.333458e-16  2.857410e-17    0.000000 -3.809881e-17   
std    1.000671e+00  1.000671e+00  1.000671e+00    1.000671  1.000671e+00   
min   -5.680625e-01 -1.760370e+00 -9.919892e-01   -0.534983 -5.411971e-01   
25%   -5.680625e-01  5.680625e-01 -9.919892e-01   -0.534983 -5.411971e-01   
50%   -5.680625e-01  5.680625e-01 -9.919892e-01   -0.534983 -5.411971e-01   
75%   -5.680625e-01  5.680625e-01  1.008076e+00   -0.534983 -5.411971e-01   
max    1.760370e+00  5.680625e-01  1.008076e+00    1.869218  1.847756e+00   

                 5             6             7           8             9   \
count  7.460000e+02  7.460000e+02  7.460000e+02  746.000000  7.460000e+02   
mean   9.524702e-18  1.523952e-16  1.714446e-16    0.000000  6.667291e-17   
std    1.000671e+00  1.000671e+00  1.000671e+00    1.000671  1.000671e+00   
min   -2.411556e-01 -5.556725e-01 -1.215897e+00   -0.448652 -1.264636e+00   
25%   -2.411556e-01 -5.556725e-01 -1.215897e+00   -0.448652 -1.264636e+00   
50%   -2.411556e-01 -5.556725e-01  8.224381e-01   -0.448652  7.907416e-01   
75%   -2.411556e-01 -5.556725e-01  8.224381e-01   -0.448652  7.907416e-01   
max    4.146700e+00  1.799621e+00  8.224381e-01    2.228901  7.907416e-01   

                 10            11            12            13            14  \
count  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02   
mean   9.524702e-17 -9.524702e-18 -1.666823e-17  4.762351e-17  2.476422e-16   
std    1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00   
min   -7.907416e-01 -2.473185e-01 -9.502953e-01 -9.375997e-01 -2.619296e+00   
25%   -7.907416e-01 -2.473185e-01 -9.502953e-01 -9.375997e-01 -7.244620e-01   
50%   -7.907416e-01 -2.473185e-01 -9.502953e-01 -9.375997e-01  1.176863e-01   
75%    1.264636e+00 -2.473185e-01  1.052304e+00  1.066553e+00  6.440290e-01   
max    1.264636e+00  4.043370e+00  1.052304e+00  1.066553e+00  2.538863e+00   

                 15            16            17            18            19  
count  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02  7.460000e+02  
mean   5.476703e-17 -2.083528e-16  2.857410e-17 -3.809881e-16 -3.143152e-16  
std    1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00  1.000671e+00  
min   -2.375219e+00 -2.700473e+00 -4.486516e-01 -2.906297e+00 -9.342128e-01  
25%   -7.540192e-01 -6.324301e-01 -4.486516e-01 -7.437078e-01 -8.409415e-01  
50%   -1.750194e-01 -1.291641e-01 -4.486516e-01 -9.243702e-03 -3.745853e-01  
75%    4.039804e-01  5.136631e-01 -4.486516e-01  8.068275e-01  5.581271e-01  
max    3.877979e+00  6.062277e+00  2.228901e+00  2.520577e+00  4.941876e+00

## Salvando variáveis

In [30]:
import pickle

In [50]:
# criar arquivo e depois atribuir var. a ele
arq1 = open('heart.pkl', 'wb')
pickle.dump(alvo,arq1)
arq2 = open('heart1.pkl', 'wb')
pickle.dump(previsores, arq2)
arq3 = open('heart2.pkl', 'wb')
pickle.dump(previsores_esc,arq3)
arq4 = open('heart3.pkl', 'wb')
pickle.dump(previsores2,arq4)
arq5 = open('heart4.pkl', 'wb')
pickle.dump(previsores3,arq5)
arq6 = open('heart5.pkl', 'wb')
pickle.dump(previsores3_esc,arq6)

In [51]:
# ler arquivo
arq1 = open('heart.pkl', 'rb')
arq2 = open('heart1.pkl', 'rb')



# executar 
previsores = pickle.load(arq2)
alvo = pickle.load(arq1)

In [49]:
# executar 
alvo = pickle.load(arq1)
alvo

array([0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0,
       0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0,
       1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1,
       1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0,
       1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1,
       1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0,

## **LEGENDA**

Age = idade (anos)

Sex = sexo (0=M; 1=F)

Chest Pain Type = tipo de dor no peito (0=TA:angina típica; 1=ATA:angina atípica; 2=NAP:dor não anginosa; 3=ASY:assintomático)

Resting BP = pressão sanguínea em repouso (mmHg)

Cholesterol = colesterol sérico (mg/dl)

Fasting BS = açúcar no sangue em jejum (mg/dl)
          0: Fasting BS < 120 mg/dl (não diabético)
          1: Fasting BS >= 120 mg/dl, (diabético)

Resting ECG = eletrocardiograma em repouso (0=Normal; 1=ST:Anormalidade da onda ST-T; 2=LVH: Hipertrofia ventricular esquerda)

Max HR = frequência cardíaca máxima

Exercise Angina = Angina induzida por exercício (0=Não; 1=Sim)

Old Peak = Depressão de ST induzida por exercício em relação ao repouso

ST_Slope = Inclinação do segmento ST (0=UP; 1=Flat; 2=Down)

Heart Disease = Doença cardíaca (0=Não possui doença cardíaca; 1=Possui doença cardíaca)

### descrição

alvo = variável que se pretende atingir (tem ou não doença cardíaca).

previsores = conjunto de variáveis previsoras com as variáveis categóricas transformadas em numéricas manualmente, sem escalonar.

previsores_esc = conjunto de variáveis previsoras com as variáveis categóricas transformadas em numéricas, escalonada.

previsores2 = conjunto de variáveis previsoras com as variáveis categóricas transformadas em numéricas pelo labelencoder.

previsores3 = conjunto de variáveis previsoras transformadas pelo labelencoder e onehotencoder, sem escalonar.

previsores3_esc = conjunto de variáveis previsoras transformadas pelo labelencoder e onehotencoder escalonada.

